### PR TITLE
Support for errors

### PR DESCRIPTION
**Pull request by gbaraldi**
This copies the method overlays from CUDA.jl. It doesn't work currently but I believe it's a nice start.

### CODE REVIEW

**Comment by brenhinkeller** [comment]
Awesome! I'll try to dig in and see how it goes!

File: src/utils.jl [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                push!(out.args, arg)
+            end
+        end
+        out
+    end
+
+    esc(rewrite(ex))
+end
+
+libcexit(x::Int32) =  @symbolcall exit(x::Int32)::Nothing
```
Comment (giordano) [inline src/utils.jl:69]:
```suggestion
libcexit(x::Int32) =  @symbolcall exit(x::Int32)::Nothing

```

File: src/quirks.jl [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+@static if VERSION <= v"1.7-"
+## prevent fallbacks to libm
+for f in (:acosh, :asinh, :atanh, :cbrt, :cosh, :exp2, :expm1, :log1p, :sinh, :tanh)
+    f_fast = Base.FastMath.fast_op[f]
+    @eval begin
+        @device_override Base.FastMath.$f_fast(x::Float32) = $f(x)
+        @device_override Base.FastMath.$f_fast(x::Float64) = $f(x)
+    end
+end
+end
```
Comment (giordano) [inline src/quirks.jl:89]:
```suggestion
end

```

File: Project.toml [hunk 3]
```diff
@@ -9,6 +9,7 @@ GPUCompiler = "61eb1bfa-7361-4325-ad38-22787b887f55"
 LLVM = "929cbde3-209d-540e-8aea-75f648917ca0"
 Libdl = "8f399da3-3557-5675-b5ff-fb832c97cbdb"
 Serialization = "9e88b42a-f829-5b0c-bbe9-9e923198166b"
+StaticTools = "86c06d3c-3f03-46de-9781-57580aa96d0a"
```
Comment (giordano) [inline Project.toml:12]: Need to add compat bound

**Comment by brenhinkeller** [comment]
See PR against your fork @gbaraldi for a fix for most of the crossed wires with the `compile` interface -- if you merge that I think it should propagate here. 

Then there's just something that's causing dynamic dispatch / allocations that didn't used to be there in some of the compiled methods in the integration tests...

**Comment by brenhinkeller** [comment]
If we can figure that out, then we just need to add some tests that test the new errors!

**Comment by MasonProtter** [comment]
Why was this merged with all tests failing?

**Comment by brenhinkeller** [comment]
The github web UI somehow did that for me when I was trying to merge the changes from the previous PR into this one to resolve conflicts... trying to revert now!

**Comment by brenhinkeller** [comment]
Ok, reverted in https://github.com/tshort/StaticCompiler.jl/commit/5e4f03febc5b8a816ef24a13c5a72019133f7a79, but looks like I can't just reopen the PR so @gbaraldi if you don't mind re-PRing from your fork... sorry for the mixup!

**Comment by brenhinkeller** [comment]
For anyone following this: reopened as #90

**Pull request merged**